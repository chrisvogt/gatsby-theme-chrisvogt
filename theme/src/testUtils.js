import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
