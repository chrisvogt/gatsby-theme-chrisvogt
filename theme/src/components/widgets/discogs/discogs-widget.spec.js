import React from 'react'
import renderer from 'react-test-renderer'

import { TestProviderWithState } from '../../../testUtils'
import DiscogsWidget from './discogs-widget'
import useSiteMetadata from '../../../hooks/use-site-metadata'

jest.mock('../../../hooks/use-site-metadata')

const mockSiteMetadata = {
  widgets: {
    discogs: {
      widgetDataSource: 'https://fake-api.example.com/widgets/discogs'
    }
  }
}

describe('Discogs Widget', () => {
  beforeEach(() => {
    useSiteMetadata.mockImplementation(() => mockSiteMetadata)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches the loading state snapshot', () => {
    const tree = renderer
      .create(
        <TestProviderWithState>
          <DiscogsWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches the loaded state snapshot', () => {
    const initialState = {
      widgets: {
        discogs: {
          state: 'SUCCESS',
          data: {
            collections: {
              releases: [
                {
                  id: 28461454,
                  instanceId: 2045415075,
                  basicInformation: {
                    id: 28461454,
                    title: 'The Rise & Fall Of A Midwest Princess',
                    year: 2023,
                    artists: [{ name: 'Chappell Roan' }],
                    cdnThumbUrl: 'https://example.com/thumb.jpg',
                    resourceUrl: 'https://discogs.com/release/123'
                  }
                }
              ]
            },
            metrics: {
              'Vinyls Owned': 37
            },
            profile: {
              profileURL: 'https://www.discogs.com/user/chrisvogt/collection'
            }
          }
        }
      }
    }

    const tree = renderer
      .create(
        <TestProviderWithState initialState={initialState}>
          <DiscogsWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches the error state snapshot', () => {
    const initialState = {
      widgets: {
        discogs: {
          state: 'FAILURE',
          data: null
        }
      }
    }

    const tree = renderer
      .create(
        <TestProviderWithState initialState={initialState}>
          <DiscogsWidget />
        </TestProviderWithState>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
