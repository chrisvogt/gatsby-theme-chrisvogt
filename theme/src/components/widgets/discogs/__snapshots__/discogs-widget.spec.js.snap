// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Discogs Widget matches the error state snapshot 1`] = `
<section
  className="css-1k28k4o"
  id="discogs"
>
  <div
    className="css-rmw57n"
  >
    <div
      className="css-59lkxq"
    >
      <h4>
        Something went wrong
      </h4>
      <p>
        Failed to load this widget.
      </p>
    </div>
    <div
      className="css-bgces8"
    />
  </div>
  <header
    className="css-x6ikgs"
  >
    <h2
      className="css-1s4ek1q"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-record-vinyl css-1f4q58s"
        data-icon="record-vinyl"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256-96a96 96 0 1 1 0 192 96 96 0 1 1 0-192zm0 224a128 128 0 1 0 0-256 128 128 0 1 0 0 256zm0-96a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Discogs
    </h2>
    <div
      className="css-1rgtmy6"
    >
      <svg
        className="svg-loaders-svg css-ao632c"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-1hw7zrq"
  >
    <div
      className="css-1giak3r"
    >
       
    </div>
    <div
      className="css-zcaudb"
    >
       
    </div>
  </div>
  <div
    className="css-sak8bt"
  >
    <div
      className="css-ccroti"
    >
      <h3
        className="css-9nh3l5"
      >
        Vinyl Collection
      </h3>
    </div>
    <p
      className="css-1qwovgy"
    >
      My owned vinyl records from Discogs.
    </p>
    <div
      className="css-1alnzao"
    >
      <div
        className="css-ngfim1"
        data-testid="vinyl-carousel"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      />
    </div>
  </div>
</section>
`;

exports[`Discogs Widget matches the loaded state snapshot 1`] = `
<section
  className="css-1k28k4o"
  id="discogs"
>
  <div
    className="css-rmw57n"
  >
    <div
      className="css-59lkxq"
    >
      <h4>
        Something went wrong
      </h4>
      <p>
        Failed to load this widget.
      </p>
    </div>
    <div
      className="css-bgces8"
    />
  </div>
  <header
    className="css-x6ikgs"
  >
    <h2
      className="css-1s4ek1q"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-record-vinyl css-1f4q58s"
        data-icon="record-vinyl"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256-96a96 96 0 1 1 0 192 96 96 0 1 1 0-192zm0 224a128 128 0 1 0 0-256 128 128 0 1 0 0 256zm0-96a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Discogs
    </h2>
    <div
      className="css-1rgtmy6"
    >
      <svg
        className="svg-loaders-svg css-ao632c"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-1hw7zrq"
  >
    <div
      className="css-1giak3r"
    >
       
    </div>
    <div
      className="css-zcaudb"
    >
       
    </div>
  </div>
  <div
    className="css-sak8bt"
  >
    <div
      className="css-ccroti"
    >
      <h3
        className="css-9nh3l5"
      >
        Vinyl Collection
      </h3>
    </div>
    <p
      className="css-1qwovgy"
    >
      My owned vinyl records from Discogs.
    </p>
    <div
      className="css-1alnzao"
    >
      <div
        className="css-ngfim1"
        data-testid="vinyl-carousel"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      />
    </div>
  </div>
</section>
`;

exports[`Discogs Widget matches the loading state snapshot 1`] = `
<section
  className="css-1pk2a5b"
  id="discogs"
>
  <header
    className="css-x6ikgs"
  >
    <h2
      className="css-1s4ek1q"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-record-vinyl css-1f4q58s"
        data-icon="record-vinyl"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256-96a96 96 0 1 1 0 192 96 96 0 1 1 0-192zm0 224a128 128 0 1 0 0-256 128 128 0 1 0 0 256zm0-96a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Discogs
    </h2>
    <div
      className="css-1rgtmy6"
    >
      <svg
        className="svg-loaders-svg css-ao632c"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-1hw7zrq"
  >
    <div
      className="css-1giak3r"
    >
       
    </div>
    <div
      className="css-zcaudb"
    >
       
    </div>
  </div>
  <div
    className="css-sak8bt"
  >
    <div
      className="css-ccroti"
    >
      <h3
        className="css-9nh3l5"
      >
        Vinyl Collection
      </h3>
    </div>
    <p
      className="css-1qwovgy"
    >
      My owned vinyl records from Discogs.
    </p>
    <div
      className="css-1alnzao"
    >
      <div
        className="css-ngfim1"
        data-testid="vinyl-carousel"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      />
    </div>
  </div>
</section>
`;
